packages/http-client-csharp/generator/Microsoft.Generator.CSharp.ClientModel/src/Models/Serialization/ScmSerializationInterfaces.cs

```diff
@@ -0,0 +1,57 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System.ClientModel.Primitives;
+using System.Collections.Generic;
+
+namespace Microsoft.Generator.CSharp.ClientModel
+{
+    internal class ScmSerializationInterfaces : SerializationInterfaces
+    {
+        private readonly CSharpType _modelType;
+        private readonly bool _isStruct;
+
+        public ScmSerializationInterfaces(TypeProvider typeProvider, bool hasJson, bool hasXml) : base(typeProvider, hasJson, hasXml)
+        {
+            _modelType = typeProvider.Type;
+            _isStruct = typeProvider.IsStruct;
+
+            if (hasJson)
+            {
+                IJsonModelTInterface = new CSharpType(typeof(IJsonModel<>), _modelType);
+                IPersistableModelTInterface = new CSharpType(typeof(IPersistableModel<>), _modelType);
+                IJsonModelObjectInterface = _isStruct ? (CSharpType)typeof(IJsonModel<object>) : null;
+                IPersistableModelObjectInterface = _isStruct ? (CSharpType)typeof(IPersistableModel<object>) : null;
+            }
+
+            // TO-DO: Handle XML serialization https://github.com/microsoft/typespec/issues/3331
+        }
+
+        protected override IReadOnlyList<CSharpType> BuildInterfaces()
```

Comment:
I think the methods should also be in this class.
In [my version of implementation](https://github.com/microsoft/typespec/pull/3313/files#diff-61618da0277b38ff06d142efc6bbd2b5aadc982c03575cb41c3a2e84d3754fd1), I make this class to build the methods
Considering: https://github.com/microsoft/typespec/pull/3355/files#r1600888643

This type should give us the 5 methods: two `Write` methods, two `Create` methods, a `GetFormatFromOptions` method.

And the snippet class gives us expression on how to serialize a property/array/dictionary/etc, but it does not give us methods. This class puts those together and give us methods.

Reply:
I've removed the builder and all the method building is done within the serialization type provider.